t/dualvar.t

```diff
@@ -131,3 +131,28 @@ SKIP: {
   ok(isdual($ary[2]), 'Is a dualvar');
 }
 
+ok !eval { dualvar() }, "arg count gets checked";
+ok !eval { dualvar(2, "a", "meep") }, "arg count gets checked";
+
+{
+  my $warning;
+  local $SIG{__WARN__} = sub { $warning = shift };
+
+  my $var = dualvar(undef, undef);
+  like($warning, qr/Use of uninitialized value in subroutine entry/, 'undef arg warning');
+  ok( isdual($var),	'Is a dualvar');
+  ok( $var == undef, 'Numeric value');
+  ok( $var eq undef, 'String value');
+
+  my $var2 = dualvar(2.2, undef);
```

Comment:
You probably want to

    undef $warning;

just before each additional test block to ensure that it's not just remaining from the previous